pennylane/devices/null_qubit.py

```diff
@@ -11,320 +11,424 @@
+    return tuple(math.zeros(s, like=interface, dtype=mp.numeric_type) for s in shape)
+
+
+@null_measurement.register
+def _(mp: ClassicalShadowMP, obj_with_wires, shots, batch_size, interface):
+    shapes = [mp.shape(obj_with_wires, Shots(s)) for s in shots]
+    if batch_size is not None:
+        shapes = [(batch_size,) + shape for shape in shapes]
+    results = tuple(math.zeros(shape, like=interface, dtype=np.int8) for shape in shapes)
+    return results if shots.has_partitioned_shots else results[0]
+
+
+@null_measurement.register
+def _(mp: CountsMP, obj_with_wires, shots, batch_size, interface):
+    outcomes = []
+    if mp.obs is None and not isinstance(mp.mv, MeasurementValue):
+        num_wires = len(obj_with_wires.wires)
+        state = "0" * num_wires
+        results = tuple({state: math.asarray(s, like=interface)} for s in shots)
+        if mp.all_outcomes:
+            outcomes = [f"{x:0{num_wires}b}" for x in range(1, 2**num_wires)]
+    else:
+        outcomes = sorted(mp.eigvals())  # always assign shots to the smallest
+        results = tuple({outcomes[0]: math.asarray(s, like=interface)} for s in shots)
+        outcomes = outcomes[1:] if mp.all_outcomes else []
+
+    if outcomes:
+        zero = math.asarray(0, like=interface)
+        for res in results:
+            for val in outcomes:
+                res[val] = zero
+    if batch_size is not None:
+        results = tuple([r] * batch_size for r in results)
+    return results[0] if len(results) == 1 else results
+
+
+@null_measurement.register(StateMP)
+@null_measurement.register(ProbabilityMP)
+def _(mp: Union[StateMP, ProbabilityMP], obj_with_wires, shots, batch_size, interface):
+    if isinstance(mp, DensityMatrixMP):
+        return _null_measurement(mp, obj_with_wires, shots, batch_size, interface)
+    wires = mp.wires or obj_with_wires.wires
+    state = [1.0] + [0.0] * (2 ** len(wires) - 1)
+    if batch_size is not None:
+        state = [state] * batch_size
+    result = math.asarray(state, like=interface)
+    return (result,) * shots.num_copies if shots.has_partitioned_shots else result
+
+
+class NullQubit(Device):
+    """Null qubit device for PennyLane. This device performs no operations involved in numerical calculations.
+       Instead the time spent in execution is dominated by support (or setting up) operations, like tape creation etc.
 
-from pennylane.ops.qubit.attributes import diagonal_in_z_basis
+    Args:
+        wires (int, Iterable[Number, str]): Number of wires present on the device, or iterable that
+            contains unique labels for the wires as numbers (i.e., ``[-1, 0, 2]``) or strings
+            (``['ancilla', 'q1', 'q2']``). Default ``None`` if not specified.
+        shots (int, Sequence[int], Sequence[Union[int, Sequence[int]]]): The default number of shots
+            to use in executions involving this device.
+
+    **Example:**
+
+    .. code-block:: python
+
+        # TODO
```

Comment:
What's the plan for this example?

Reply:
just put some stuff kinda matching default.qubit, plus something to highlight tracking resources since it's a feature we'd highlight to external users